pkgs/development/web/nodejs/nodejs.nix

```diff
@@ -128,25 +128,71 @@ let
 
     inherit patches;
 
-    doCheck = lib.versionAtLeast version "16"; # some tests fail on v14
+    __darwinAllowLocalNetworking = true; # for tests
+
+    # TODO: what about tests when cross-compiling?
+    # Note that currenly stdenv does not run check phase if build ≠ host.
+    doCheck = true;
 
     # Some dependencies required for tools/doc/node_modules (and therefore
     # test-addons, jstest and others) target are not included in the tarball.
     # Run test targets that do not require network access.
-    checkTarget = lib.concatStringsSep " " [
+    checkTarget = lib.concatStringsSep " " ([
       "build-js-native-api-tests"
       "build-node-api-tests"
       "tooltest"
       "cctest"
+    ] ++ lib.optionals (lib.versionAtLeast version "20") [
```

Comment:
It seems to be Darwin-specific, no? If so, could we skip it only for 18.x on Darwin?

Reply:
Seems to build fine with checks on Linux, yes. Changed that to check that we are not building v18 on Darwin.